### PR TITLE
is0403: add peer to peer tests and update is0401 txt checks

### PR DESCRIPTION
**Pull request by andrewbonney**

### CODE REVIEW

**Comment by andrewbonney** [comment]
Just wondering if it's worth including a change to the final return statement in test_12 of is0401. This would be a WARNING rather than FAIL, and say 'No matching mDNS announcement found for Node, however this will not affect operation in registered mode', given the spec discussions around the need for the Node mDNS advertisement. Thoughts?

**Comment by garethsb-sony** [comment]
> Just wondering if it's worth including a change to the final return statement in test_12 of is0401. This would be a WARNING rather than FAIL, and say ...

Yes, sounds good - do we have a wiki section to link to? :-)

**Comment by andrewbonney** [comment]
Done. I've only included the link against the 'worst case' failures, but hopefully that's enough.

File: IS0403Test.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            port = node.port
+            if address in self.node_url and ":{}".format(port) in self.node_url:
+                properties = self.convert_bytes(node.properties)
+                for ver_txt in ["ver_slf", "ver_src", "ver_flw", "ver_dvc", "ver_snd", "ver_rcv"]:
+                    if ver_txt not in properties:
+                        return test.FAIL("No '{}' TXT record found in Node API advertisement.".format(ver_txt))
+                    try:
+                        version = int(properties[ver_txt])
+                        if version < 0:
+                            return test.FAIL("Version ('{}') TXT record must be greater than zero.".format(ver_txt))
```
[the remark this answers is not in the document]
Comment (andrewbonney) [reply IS0403Test.py]: Oops. Fixed

**Comment by andrewbonney** [comment]
@garethsb-sony would you mind re-approving following that fix?

**Pull request merged**